### PR TITLE
r2snowman - added just-function decompilation mode

### PR DESCRIPTION
**Pull request by s0i37**
After commit https://github.com/yegord/snowman/pull/164 the snowman can decompile only one function.

### CODE REVIEW

File: r2snowman/r2snow [hunk 1]
```diff
@@ -1,18 +1,15 @@
-#!/bin/sh
-TARGET="$1"
+#!/usr/bin/python
+import r2pipe
+from pygments import highlight
+from pygments.lexers import CppLexer
+from pygments.formatters import Terminal256Formatter
+from os import popen
 
-if [ -z "$1" ]; then
```
Comment (radare) [inline r2snowman/r2snow]: why remove the help? you are removing the ability to use r2snow from the commandline, check for this if r2pipe.open() fails

File: r2snowman/r2snow.pl [hunk 2]
```diff
@@ -1,62 +0,0 @@
...
-  return 0;
-}
-
-initNewlines();
-
-foreach my $a (@addrof) {
-  my ($addr, $offset) = split(" ", $a);
-  my $file = R2SNOW_SOURCE . ".txt";
-  my $line = lineFor($offset);
-  print "CL $file:$line $addr\n";
```
Comment (radare) [inline r2snowman/r2snow.pl]: you are also removing the ability to import the decompilation into r2

File: r2snowman/README.md [hunk 3]
```diff
@@ -9,6 +9,6 @@ r2snowman
 - - ----(    . )---\|---
```
Comment (radare) [inline r2snowman/README.md]: fix the ascii art
screenshot 2018-10-25 at 13 41 58

File: r2snowman/r2snow.pl (hunk 2 above)
Comment (s0i37) [reply r2snowman/r2snow.pl]: Hmm. I noticed that previously r2 used "feature/output-ranges" branch of snowman. But just-function decompilation implemented only in main branch. So I need to make one more pool request to this branch

**Comment by radare** [comment]
just update this pr

**Comment by s0i37** [comment]
Yes. That I need to do? Update help menu?

**Comment by Maijin** [comment]
Yes @s0i37

**Comment by radare** [comment]
solve conflict spls

**Comment by radare** [comment]
Why did u changed the ascii art? The pr is removing some features

**Comment by Maijin** [comment]
Because the ASCII art was the conflict

**Comment by radare** [comment]
wtf

**Comment by s0i37** [comment]
I think that this solution has lost relevance. r2dec and ghidra have already just-function decompilation mode.

**Comment by radare** [comment]
well, r2snowman should work in the same way as r2dec and r2ghidra, in fact we may evolve all the decompilers to support all the features in the same way with r2

**Comment by s0i37** [comment]
The main problem - I made changes of snowman in main branch, but r2 used another branch `feature/output-ranges`. So I need to make feature request in this branch again. But now I have not enoght time.
Moreover I don't fully understand that I need to do yet about "ASCII art was the conflict"

**Comment by radare** [comment]
> On 7 Oct 2019, at 11:37, s0i37 <notifications@github.com> wrote:
> 
> The main problem - I made changes of snowman in main branch, but r2 used another branch feature/output-ranges. So I need to make feature request in this branch again. But now I have not enoght time.
> Moreover I don't fully understand that I need to do yet about "ASCII art was the conflict"
> 
> 

which ascii art? the repo was moved, time is the only issue here from my pov. also i see snowman pretty abandoned, and having retdec, ghidra and r2dec i dont see much incentive in spending more time on making an x86-only decompiler work here..

**Comment by radare** [comment]
the pr must be reopened against radareorg/ otherwise the conflicts dont work

**Pull request closed without merging**